modules/packages/CopyAggregation.chpl

```diff
@@ -0,0 +1,472 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+   .. warning::
+     This module represents work in progress. The API is unstable and likely to
+     change over time.
+
+   This module provides an aggregated version of copy/assignment for trivially
+   copyable types. Trivially copyable types require no special behavior to be
+   copied and merely copying their representation is sufficient. They include
+   ``numeric`` and ``bool`` types as well as tuples or records consisting only
+   of ``numeric``/``bool``. Records cannot have user-defined copy-initializers,
+   deinitializers, or assignment overloads.
+
+   Aggregated copies can provide a significant speedup for batch assignment
+   operations that do not require ordering of operations. The results of
+   aggregated operations are not visible until the aggregator is deinitialized
+   or an explicit ``flush()`` call is made. The current implementation requires
+   that one side of the copy is always local. When the source is always local
+   and the destination may be remote a :record:`DstAggregator` should be used.
+   When the destination is always local and the source may be remote, a
+   :record:`SrcAggregator` should be used. Note that aggregators are not
+   parallel safe and are expected to be created on a per-task basis. Each
+   aggregator has a per-peer buffer, so memory usage increases with number of
+   locales. The following example demonstrates  using :record:`SrcAggregator`
+   to reverse an array:
+
+   .. code-block:: chapel
+
+     use BlockDist, CopyAggregation;
+
+     const size = 10000;
+     const space = {0..size};
+     const D = space dmapped Block(space);
+     var A, reversedA: [D] int = D;
+
+     forall (rA, i) in zip(reversedA, D) with (var agg = new SrcAggregator(int)) do
+       agg.copy(rA, A[size-i]);
+
+     // no flush required, flushed when aggregator is deinitialized
+
+     forall (rA, i) in zip(reversedA, D) do
+       assert(rA == size-i);
+
+
+   It's important to be aware that aggregated operations are not consistent with
+   regular operations and updates may not be visible until the aggregator is
+   deinitialized or an explicit call to ``flush()`` is made.
+
+   .. code-block:: chapel
+
+     use CopyAggregation;
+
+     var a = 0;
+     on Locales[numLocales-1] {
+       var agg = new DstAggregator(int);
+       var b = 1;
+       agg.copy(a, b);
+       writeln(a);   // can print 0 or 1
+       agg.flush();
+       writeln(a);   // must print 0
```

Comment:
"must print 1"?

Reply:
Thank you! Fixed in https://github.com/chapel-lang/chapel/pull/19426

(This was originally from https://github.com/chapel-lang/chapel/blob/110feff72e0eb860d2cdff7b32aa5d55f760e823/modules/packages/UnorderedCopy.chpl#L60-L69, but I wanted to show a DstAggregator instead, but missed updating this comment.)